src/metrics.ts

```diff
@@ -129,79 +180,62 @@ export class MetricsProcessor implements MetricsProcessorInterface {
     const metricsData: MetricsData[] = [];
 
     // clone map and clear data
-    const clonedData = new Map(this.data);
-    this.data.clear();
-
-    for (const event of clonedData.values()) {
-      if (event.target && !event.target.anonymous) {
-        let targetAttributes: KeyValue[] = [];
-        if (event.target.attributes) {
-          targetAttributes = Object.entries(event.target.attributes).map(
-            ([key, value]) => {
-              const stringValue =
-                value === null || value === undefined
-                  ? ''
-                  : this.valueToString(value);
-              return { key, value: stringValue };
-            },
-          );
-        }
-
-        let targetName = event.target.identifier;
-        if (event.target.name) {
-          targetName = event.target.name;
-        }
-
-        const td: TargetData = {
-          identifier: event.target.identifier,
-          name: targetName,
-          attributes: targetAttributes,
-        };
-        targetData.push(td);
-      }
+    const clonedEvaluationAnalytics = new Map(this.evaluationAnalytics);
+    const clonedTargetAnalytics = new Map(this.targetAnalytics);
+    this.evaluationAnalytics.clear();
+    this.targetAnalytics.clear();
+    this.evaluationAnalyticsExceeded = false;
+    this.targetAnalyticsExceeded = false;
 
+    clonedEvaluationAnalytics.forEach((event) => {
       const metricsAttributes: KeyValue[] = [
         {
           key: FEATURE_IDENTIFIER_ATTRIBUTE,
           value: event.featureConfig.feature,
         },
-        {
-          key: FEATURE_NAME_ATTRIBUTE,
-          value: event.featureConfig.feature,
-        },
         {
           key: VARIATION_IDENTIFIER_ATTRIBUTE,
           value: event.variation.identifier,
         },
-        {
-          key: SDK_TYPE_ATTRIBUTE,
-          value: SDK_TYPE,
-        },
-        {
-          key: SDK_LANGUAGE_ATTRIBUTE,
-          value: SDK_LANGUAGE,
-        },
-        {
-          key: SDK_VERSION_ATTRIBUTE,
-          value: VERSION,
-        },
-        {
-          key: TARGET_ATTRIBUTE,
-          value: event?.target?.identifier ?? null,
-        },
+        { key: FEATURE_NAME_ATTRIBUTE, value: event.featureConfig.feature },
+        { key: SDK_TYPE_ATTRIBUTE, value: SDK_TYPE },
+        { key: SDK_LANGUAGE_ATTRIBUTE, value: SDK_LANGUAGE },
+        { key: SDK_VERSION_ATTRIBUTE, value: VERSION },
+        { key: TARGET_ATTRIBUTE, value: event?.target?.identifier ?? null },
       ];
 
-      // private target attributes
-      // need more info
-
       const md: MetricsData = {
         timestamp: Date.now(),
         count: event.count,
         metricsType: MetricsDataMetricsTypeEnum.Ffmetrics,
         attributes: metricsAttributes,
       };
       metricsData.push(md);
-    }
+    });
+
+    clonedTargetAnalytics.forEach((target) => {
```

Comment:
Same here. Also totally a non-issue.
```suggestion
    const targetData: TargetData[] = clonedTargetAnalytics.map((target) => {
```